File: ROT13Cipher.py

def encrypt(pt):
    encpt1=list(pt)
    for i in range(len(pt)):
        n=ord(pt[i])
        if(n>=78):
            n+=13
            n=n-90+64
            ch=chr(n)
            encpt1[i]=ch
        else:
            n+=13
            ch=chr(n)
            encpt1[i]=ch
    return (''.join(encpt1))


def decrypt(encpt):
    decpt1=list(encpt)
    for i in range(len(encpt)):
        n=ord(encpt[i])
        if(n<=77):
            n-=13
            n=91-(65-n)
            ch=chr(n)
            decpt1[i]=ch
        else:
            n-=13
            ch=chr(n)
            decpt1[i]=ch
    return (''.join(decpt1))


pt = input("Enter plaintext: ")
encpt = encrypt(pt)
print("Encrypted text: ",encpt)
decpt = decrypt(encpt)
print("Decrypted text: ",decpt)

File: ROT13Cipher.java
import java.util.Scanner;
public class ROT13Cipher{

	static String encrypt(String pt){
		int i,n;
		char ch;
		char[] encpt1=pt.toCharArray();
		for(i=0;i<pt.length();i++){
			n=encpt1[i];
			if(n>=78){
				n+=13;
				n=n-90+64;
				ch=(char)n;
				encpt1[i]=ch;
			}
			else{
				n+=13;
				ch=(char)n;
				encpt1[i]=ch;
			}
		}
		String encpt2=new String(encpt1);
		return encpt2;
	}

	static String decrypt(String encpt){
		int i,n;
		char ch;
		char[] decpt1=encpt.toCharArray();
		for(i=0;i<encpt.length();i++){
			n=decpt1[i];
			if(n<=77){
				n-=13;
				n=91-(65-n);
				ch=(char)n;
				decpt1[i]=ch;
			}
			else{
				n-=13;
				ch=(char)n;
				decpt1[i]=ch;
			}
		}
		String decpt2=new String(decpt1);
		return decpt2;
	}

	public static void main(String[] args){
		String pt,encpt,decpt;
		System.out.println("Enter plaintext: ");
		Scanner scanner=new Scanner(System.in);
		pt=scanner.nextLine();
		encpt=encrypt(pt);
		System.out.println("Encrypted text: ");
		System.out.println(encpt);
		decpt=decrypt(encpt);
		System.out.println("Decrypted text: ");
		System.out.println(decpt);
	}
}